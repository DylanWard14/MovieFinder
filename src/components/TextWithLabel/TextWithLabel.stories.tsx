import React from 'react';
import { withKnobs, text, select } from '@storybook/addon-knobs';
import styled from 'styled-components/macro';

import TextWithLabel from './TextWithLabel';

const Wrapper = styled.div`
    width: 100%;
    height: 100%;
    background: black;
`;

export default {
    title: 'Components/TextWithLabel',
    decorators: [(storyFn: any) => <Wrapper>{storyFn()}</Wrapper>, withKnobs],
};

export const DefaultExample = () => <TextWithLabel label="Label" content="This is some content" />;
export const SmallLabelExample = () => (
    <TextWithLabel label="Label" content="This is some content" labelSize="small" contentSize="large" />
);
export const PropsExample = () => (
    <TextWithLabel
        label={text('label', 'label')}
        content={text('content', 'this is some content')}
        labelSize={select('labelSize', ['small', 'large'], 'large')}
        contentSize={select('contentSize', ['small', 'large'], 'small')}
    />
);
