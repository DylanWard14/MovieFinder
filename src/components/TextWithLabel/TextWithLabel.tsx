import React from 'react';
import styled from 'styled-components/macro';

const Component = styled.div`
    display: flex;
    flex-direction: column;
    margin-bottom: 1rem;
`;

const Heading = styled.p<{ labelSize: SizeType }>`
    ${props => (props.labelSize === 'large' ? props.theme.typography.largeLabel : props.theme.typography.smallLabel)}
`;

const Content = styled.p<{ contentSize: SizeType }>`
    ${props => (props.contentSize === 'large' ? props.theme.typography.largeLabel : props.theme.typography.smallLabel)}
`;

type SizeType = 'small' | 'large';

interface TextWithLabelProps {
    label: string;
    content: string;
    labelSize?: SizeType;
    contentSize?: SizeType;
    className?: string;
}

const TextWithLabel: React.FunctionComponent<TextWithLabelProps> = ({
    label,
    content,
    labelSize = 'large',
    contentSize = 'small',
    className,
}) => {
    return (
        <Component className={className}>
            <Heading labelSize={labelSize}>{label}</Heading>
            <Content contentSize={contentSize}>{content}</Content>
        </Component>
    );
};

export default TextWithLabel;
