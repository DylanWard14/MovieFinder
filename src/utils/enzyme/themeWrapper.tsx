import React from 'react';
import { render, shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';

export const renderWrapper = (component: React.ReactNode) => {
    return render(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const shallowWrapper = (component: React.ReactNode) => {
    return shallow(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const mountWrapper = (component: React.ReactNode) => {
    return mount(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};
