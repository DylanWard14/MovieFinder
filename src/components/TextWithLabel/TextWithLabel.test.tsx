import React from 'react';

import { renderWrapper } from '../../utils/enzyme/themeWrapper';
import TextWithLabel from './TextWithLabel';

describe('TextWithLabel', () => {
    it('renders with a label and content correctly', () => {
        const wrapper = renderWrapper(<TextWithLabel label="label" content="content" />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders with a label and content and sizes set correctly', () => {
        const wrapper = renderWrapper(
            <TextWithLabel label="label" content="content" labelSize="small" contentSize="large" />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
