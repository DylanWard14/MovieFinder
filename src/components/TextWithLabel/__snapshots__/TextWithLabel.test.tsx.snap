// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithLabel renders with a label and content and sizes set correctly 1`] = `
<div
  class="TextWithLabel__Component-sc-13uehc1-0 juyjpy"
>
  <p
    class="TextWithLabel__Heading-sc-13uehc1-1 dvvKXU"
  >
    label
  </p>
  <p
    class="TextWithLabel__Content-sc-13uehc1-2 ZPJpN"
  >
    content
  </p>
</div>
`;

exports[`TextWithLabel renders with a label and content correctly 1`] = `
<div
  class="TextWithLabel__Component-sc-13uehc1-0 juyjpy"
>
  <p
    class="TextWithLabel__Heading-sc-13uehc1-1 bTEzLT"
  >
    label
  </p>
  <p
    class="TextWithLabel__Content-sc-13uehc1-2 djIfsi"
  >
    content
  </p>
</div>
`;
